### PR TITLE
update roadmap after v0.5

### PR DESCRIPTION
**Pull request by bassam**

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -2,28 +2,20 @@
...
- - TPRs for object store and file system
- - Initial Rook Upgrade support
-
 ## Rook 0.6 (September 2017)
 
  - Ceph Block storage declared Beta on K8S
  - Kubernetes 1.8 Support
  - Rook volume plugin support in K8S
+ - Rook agent on client nodes
  - Rook can upgrade itself
```
Comment (travisn) [inline ROADMAP.md]: This sounds like upgrades are fully automated, but we're just saying that it's possible to upgrade. Can we phrase this "Rook is upgradable"?

File: ROADMAP.md [hunk 2]
```diff
@@ -2,28 +2,20 @@
...
- - Improved data placement and pool configuration
+ - Support for local storage (and arbitrary volumes) underneath OSD pods
  - Support for public cloud environments
- - Performance Testing infra
+ - Improved test matrix and more test coverage
```
Comment (travisn) [inline ROADMAP.md]: This isn't really about adding new tests, but empowering development. Can we phrase this: "Dev team can efficiently add end to end tests"?

File: ROADMAP.md [hunk 3]
```diff
@@ -2,28 +2,20 @@
...
+ - CRDs for object storage and file system
```
Comment (travisn) [inline ROADMAP.md]: This doesn't belong under the goal of "block storage in beta". Should this be moved to 0.7? I'm not sure we should block 0.6 for this. There are some details to work out here like what it means to have multiple object stores and file systems.

Comment (travisn) [inline ROADMAP.md]: Really the goal is that we are 100% kubernetes api and our own api service is just an optional component (for standalone or rookctl scenarios)

File: ROADMAP.md [hunk 4]
```diff
@@ -2,28 +2,20 @@
...
- - Backup of the cluster config and data required to recover it
- - Publish documentation to rook.io
- - Upgrade to Ceph Luminous and make the default
- - Long-haul testing infra
- - TPRs for object store and file system
- - Initial Rook Upgrade support
-
 ## Rook 0.6 (September 2017)
 
  - Ceph Block storage declared Beta on K8S
```
Comment (travisn) [inline ROADMAP.md]: Would be good to define what is understood by beta, though maybe it is implicit in this list

File: ROADMAP.md [hunk 5]
```diff
@@ -2,28 +2,20 @@
...
  - Kubernetes 1.8 Support
  - Rook volume plugin support in K8S
+ - Rook agent on client nodes
```
Comment (travisn) [inline ROADMAP.md]: Shall we expand on why we need an agent? Is this just part of the flex volume work or shall we expand on its purpose?

Comment (bassam) [inline ROADMAP.md]: @jbw976 design doc will have the details.

File: ROADMAP.md (hunk 2 above)
Comment (bassam) [inline ROADMAP.md]: my point is that dev will improve the test coverage

**Pull request merged**